### PR TITLE
unlock db

### PR DESCRIPTION
**Pull request by lainio**
- new pkgs x for thread-safety and http for graceful shutdown
- graceful shutdown for all managed db instances
- TLS, docs, http pkg for gracefull shutdown
- missing error check, try.To call
- db name for logs
- support for mem db name, and fixed err handling in GracefulStop
- gofmt
- err2 update

### CODE REVIEW

File: crypto/db/handle.go [hunk 1]
```diff
@@ -11,4 +14,10 @@ type Handle interface {
 	Backup() (did bool, err error)
 	Wipe() (err error)
 	Close() (err error)
+	SetStatusFn(f OnFn)
```
Comment (lauravuo) [inline crypto/db/handle.go:17]: How is this used/who uses it?

Comment (lainio) [reply crypto/db/handle.go:17]: The bolt and mem DB implement it, **even if it's not currently used**. It's also something that we *might remove after we have discussed it*. So I left it there so that we can do just that :-D

## TL;DR
Instead, to have an on/off (enabled/disabled) flag in the DB instance, we have an `onFn` function that allows more and handles thread safety, etc.

## Whole Story
For instance, I could use it during the graceful shutdown. Still, then I thought to keep it simple and not implement or call it yet because nothing should happen when all the networking is in shutdown mode, **and disabling the DB at the beginning of the shutdown is stupid because there can be something to finish**.

The idea for it is that it allows a very efficient way to implement a process level flag to, e.g., disallow any new DB connections, e.g., we could have a short service break by following:
1. Set `disallow = true`. `disallow` would be `atomic.Bool`, and the `SetStatusFn` would get the function that returns the flag's value. Of course, we should call the `SetStatusFn` for every DB instance.
2. Closes all open DBs
3. Do whatever is needed and meanwhile no DB can be opened. That's what it would protect, for example.
4. Set `disallow = false`
5. Next request to use any DB would succeed.

**Comment by lauravuo** [comment]
LGTM!

**Review by lauravuo — APPROVED** [review]
